src/librustc_mir/interpret/place.rs

```diff
@@ -585,8 +586,9 @@ where
         use rustc::mir::StaticKind;
 
         Ok(match place_static.kind {
-            StaticKind::Promoted(promoted, _) => {
-                let instance = self.frame().instance;
+            StaticKind::Promoted(promoted, promoted_substs) => {
+                let substs = self.frame().instance.substs.subst(*self.tcx, promoted_substs);
```

Comment:
Use `subst_from_frame_and_normalize_erasing_regions` instead of handrolling the subst call